### PR TITLE
Use unique work directories when building with docker.

### PR DESCRIPTION
**Pull request by drewmoseley**
This allows multiple builds to be run in parallel.

Changelog: Title
Signed-off-by: Drew Moseley <drew.moseley@northern.tech>

# External Contributor Checklist

🚨 Please review the [guidelines for contributing](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md) to this repository.

- [ ] Make sure that all commits have a [`Changelog`](https://github.com/mendersoftware/mender/blob/master/CONTRIBUTING.md#changelog-tags) tag. If nothing should be added to the Changelog, add a `Changelog: None` tag. If there is a change, add `Changelog: Commit`, or `Changelog: Title`, depending on what should be included in the changelog.

- [ ] Make sure that all commits are signed with [`git --signoff`](https://git-scm.com/book/en/v2/Git-Tools-Signing-Your-Work). Also note that the signoff author must match the author of the commit.

### Description

Please describe your pull request.

Thank you!

### CODE REVIEW

**Comment by drewmoseley** [comment]
Looks like this requires "sudo" in the host OS.

File: docker-mender-convert [hunk 1]
```diff
@@ -25,13 +25,21 @@ MENDER_CONVERT_DIR="$(pwd)"
...
   --privileged=true \
   --cap-add=SYS_MODULE \
   -v /dev:/dev \
   -v /lib/modules:/lib/modules:ro \
   --env MENDER_ARTIFACT_NAME=${MENDER_ARTIFACT_NAME} \
   --env MENDER_CONVERT_VERSION=${MENDER_CONVERT_VERSION} \
   $IMAGE_NAME "$@"
+
+mv ${WORK_DIR}/convert.log ./convert.log.${WORK_SUFFIX}
+sudo rm -rf ${WORK_DIR}
```
Comment (kacf) [inline docker-mender-convert:45]: This step can be done inside docker instead to avoid the need for sudo.

**Review by lluiscampos — COMMENTED** [review]
One disadvantage that I can see is that the log is lost (say, hidden) when the tool fails and aborts half way. At least we echo at the beginning which is the workdir being used, so that on failures it can be inspected manually.

**Comment by kacf** [comment]
> One disadvantage that I can see is that the log is lost (say, hidden) when the tool fails and aborts half way. At least we echo at the beginning which is the workdir being used, so that on failures it can be inspected manually.

Then I suggest putting the log in a different directory, either the deploy directory, or a dedicated volume for logs. Then make sure it has the same owner as the directory containing it.

File: docker-mender-convert (hunk 1 above)
Comment (drewmoseley) [reply docker-mender-convert:45]: The _contents_ of WORK_DIR can be removed but not work.XXXX in the parent.

**Comment by drewmoseley** [comment]
I reworked a bit to address the above comments and hopefully fix the CI failures.

**Comment by drewmoseley** [comment]
@kacf @lluiscampos @mirzak  any thoughts on the latest version of this? It's going out to a customer soon and if it's merged it's one less step for me.

**Review by oleorhagen — APPROVED** [review]
LGTM

**Review by kacf — APPROVED** [review]

**Comment by kacf** [comment]
Yes, sorry, this got a bit lost.

**Pull request merged**